okhttp/src/main/java/com/squareup/okhttp/internal/spdy/SpdyConnection.java

```diff
@@ -791,90 +833,51 @@ public void pushPromise(int streamId, int promisedStreamId, List<Header> request
     }
   }
 
-  /** Even, positive numbered streams are pushed streams in HTTP/2. */
+  /** Even, positive numbered streams are pushed streams in HTTP/2 and SPDY/3. */
   private boolean pushedStream(int streamId) {
-    return protocol == Protocol.HTTP_2 && streamId != 0 && (streamId & 1) == 0;
+    return (protocol == Protocol.HTTP_2 || client) && streamId != 0 && (streamId & 1) == 0;
```

Comment:
This change seems to be incorrect - according to the SPDY specification, only the server can create push streams, and any streams started by the client must have an odd stream ID. 

https://www.chromium.org/spdy/spdy-protocol/spdy-protocol-draft3-1#TOC-2.3.2-Stream-creation